### PR TITLE
[Network]Support "--custom-routes" argument for "az network vnet-gateway create/update"

### LINKED ISSUE #9327: Virtual network gateway custom routes feature
**Resource Provider**
What is the Azure resource provider your feature is part of?
Network

**Description of Feature or Work Requested**
Provide a brief description of the feature or work requested. A link to conceptual documentation may be helpful too.
This feature is added for customers to specify Custom routes (list of Ip addresses) on Virtual network gateway to have access to.

**Minimum API Version Required**
What is the minimum API version of your service required to implement your feature?
2019-02-01 

**Swagger Link**
Provide a link to the location of your feature(s) in the REST API specs repo. This should be on the master branch
of the REST API specs repo.
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/virtualNetworkGateway.json

**Target Date**
If you have a target date for release of this feature/work, please provide it. While we can't guarantee these dates,
it will help us prioritize your request against other requests.
This feature is already released.

### PR DESCRIPTION
**Pull request by MyronFanQiu**
Fixes #9327 
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 1]
```diff
@@ -1200,6 +1200,7 @@ def load_arguments(self, _):
         c.argument('radius_server', min_api='2017-06-01', help='Radius server address to connect to.', arg_group='VPN Client')
         c.argument('radius_secret', min_api='2017-06-01', help='Radius secret to use for authentication.', arg_group='VPN Client')
         c.argument('client_protocol', min_api='2017-06-01', help='Protocols to use for connecting', nargs='+', arg_group='VPN Client', arg_type=get_enum_type(VpnClientProtocol))
+        c.argument('custom_routes', help='Space-separated list of CIDR prefixes representing the custom routes address space specified by the customer for virtual network gateway and VpnClient.', nargs='+', arg_group='VPN Client')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/network/_params.py]: Does it have min_api?

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/network/_params.py]: good catch!

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -10,6 +10,10 @@ Release History
 * [BREAKING] When not specified, the default value for `--start-task-wait-for-success` on `az batch pool create` is now true (was false).
 * [BREAKING] The default value for Scope on AutoUserSpecification is now always Pool (was Task on Windows nodes, Pool on Linux nodes). This argument is not exposed via the commandline, but can be set in the `--json-file` arguments.
 
+**Network**
+
+* az network vnet-gateway create/update: Fixes #9327. Support `--custom-routes` argument to set custom routes address space for vnet gateway and vpn client.
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst]: VNet and VPN

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 3]
```diff
@@ -1213,6 +1213,7 @@ def load_arguments(self, _):
         c.argument('radius_server', min_api='2017-06-01', help='Radius server address to connect to.', arg_group='VPN Client')
         c.argument('radius_secret', min_api='2017-06-01', help='Radius secret to use for authentication.', arg_group='VPN Client')
         c.argument('client_protocol', min_api='2017-06-01', help='Protocols to use for connecting', nargs='+', arg_group='VPN Client', arg_type=get_enum_type(VpnClientProtocol))
+        c.argument('custom_routes', min_api='2019-02-01', help='Space-separated list of CIDR prefixes representing the custom routes address space specified by the customer for virtual network gateway and VpnClient.', nargs='+', arg_group='VPN Client')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/network/_params.py]: I have two questions about the role/function of custom_routes. What is the difference between address-prefixes and custom roues? It looks that custom routes not only work on VPN client, but also virtual network gateway. In this way, why do you want to place it in VPN Client Arguments group?

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/network/_params.py]: In my understanding, there is a pair relationship between VPN client and virtual network gateway. You can set the `custom_routes` in the virtual network gateway, then VPN client can use it. If you check the commands under `az network vnet-gateway vpn-client`, there are just two commands `generate` and `show-url` without any CRUD commands.

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/network/_params.py]: address-prefixes is used for P2S VPN client. Although they are using same structure, the content or usage are different.

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/network/_params.py]: so. Based on the SDK's description, it's used by VPN client.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 4]
```diff
@@ -4100,6 +4104,13 @@ def update_vnet_gateway(cmd, instance, sku=None, vpn_type=None, tags=None,
     if enable_bgp is not None:
         instance.enable_bgp = enable_bgp.lower() == 'true'
 
+    if custom_routes and cmd.supported_api_version(min_api='2019-02-01'):
+        if not instance.custom_routes:
+            instance.custom_routes = AddressSpace()
+        if not instance.custom_routes.address_prefixes:
+            instance.custom_routes.address_prefixes = []
```
[the remark this answers is not in the document]
Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/network/custom.py]: done

**Pull request merged**